### ISSUE TITLE
comformability issue

### ISSUE THREAD

**Issue by ealda** [bquistorff/synth_runner#4]
thanks for this fantastic command. I ran your algorithm with my dataset and I ran into a comformability issue. 
 I checked online and I noticed that synth also run into some of the same issues and someone tweaked the code to fix it. I am not sure, however, if you have received emails about this issue.
Your kind advice will be greatly appreciated

**Comment by bquistorff** [comment]
If you estimate a single synthetic control model do you know where the error is? You can turn on tracing (`help trace`) to find that out.

You mention that someone tweaked the code to fix it. Can you provide more details on that?

I've run across a few errors with `synth`.

**Comment by ealda** [comment]
Thanks for your response. They tweaked the synth code not synth_runner. Apologies for misunderstanding. 
Will run code and post
thanks

**Comment by ealda** [comment]
- synth `anything', keep(`keep') trperiod(`trperiod') trunit(`trunit') `options'
= synth wdi_homicide    percentmale  popdens  wdi_homicide(2012) wdi_homicide(2010)	wdi_homicide(2008)	wdi_homicide(2006)	wdi_homicide(2004)	wdi_homicide(2002)	wdi_homicide(2000),	keep(/var/folders/l_/3xqwhm2x14j_nt121b6h
> 15_c0000gn/T//S_01212.000002) trperiod(2013) trunit(53) replace
conformability error


I am running version 1.1.5. The error shows up after it starts the synth command. 
Apologies if code does not look good

**Comment by ealda** [comment]
I figured out the problem
it is with the `synth_wrapper.ado` file. Because `synth' shows conformability errors, the tweak in the non critical command fixes the bug shown in the stata forum 
http://www.statalist.org/forums/forum/general-stata-discussion/general/184880-synth-conformability-error  The new `.ado' file is named `synth2'. Thus, in order for the `synth_wrapper.ado' file to work I tweaked the command `synth `anything', keep(`keep') trperiod(`trperiod') trunit(`trunit') `options'' to 
`synth2 `anything', keep(`keep') trperiod(`trperiod') trunit(`trunit') `options'' and it seems to be working

**Comment by bquistorff** [comment]
That seems like a good solution. Thanks for noting that here.

**Comment by GAlbuquerque** [comment]
Sorry for reopening this, but I am having the same problem descrived by ealda. I followed the instructions about changing synth's ado and synth_wrapper. However, the problem remains exact the same. 

More strange information: I already used synth_runner sucessfully to calculate the p-value of  individual's units effects. However, when I tried to do the same for all these units at once in order to calculate the avarage effect's p-value I have the conformability issue. Do any of you have know how to help?

Thank you in advance

**Comment by ealda** [comment]
@GAlbuquerque This is what I did with the code to make it run:

For the `synth.ado` of package `synth'
in line 3:
change: program `synth, eclass`
for
program `synth2 , eclass`
save it as `synth2.ado`


in line 35 of `synth_wrapper.ado`, I changed:
`synth `anything', keep(`keep') trperiod(`trperiod') trunit(`trunit') `options'`

for

 `synth2 `anything', keep(`keep') trperiod(`trperiod') trunit(`trunit') `options'`

save it as 

`synth_wrapper.ado`

I hope it works
Best

**Comment by bquistorff** [comment]
GAlbuquerque, if you data is shareable you can send me a minimal working example.

**Comment by GAlbuquerque** [comment]
ealda, thank you for your detailed intructions. Before asking I did exactly what you recomend, without any success. So I decided to redo my code from the scratch and I no longer have the conformability issue and the program could run without problems for some time. 

Unfortunatelly now I have a different problem. I will try harder to get a solution myself before openning a new issue and bothering you two again.

bquistorff, thank you for your offer to help.
